app/services/source_refresh_service.rb

```diff
@@ -0,0 +1,48 @@
+class SourceRefreshService
+  def initialize(source_id)
+    @source = Source.find(source_id)
+  end
+
+  def process
+    if @source.refresh_task_id.nil?
+      dispatch_refresh_upload_task
+    else
+      task = Task.find(@source.refresh_task_id)
+
+      if task.state == "completed"
+        if task.child_task_id.nil?
```

Comment:
@hsong-rh If the task.status has en error it would have never created a child task and this logic would report that we are still running the refresh and we can't start one.